ambry-api/src/main/java/com.github.ambry/rest/RequestPath.java

```diff
@@ -106,16 +119,18 @@ public static RequestPath parse(String path, Map<String, Object> args, List<Stri
       operationOrBlobId = args.get(RestUtils.Headers.BLOB_ID).toString();
     }
 
-    return new RequestPath(prefixFound, clusterNameFound, pathAfterPrefixes, operationOrBlobId, subResource);
+    return new RequestPath(prefixFound, clusterNameFound, pathAfterPrefixes, operationOrBlobId, subResource,
+        blobSegmentNum);
   }
 
   RequestPath(String prefix, String clusterName, String pathAfterPrefixes, String operationOrBlobId,
-      RestUtils.SubResource subResource) {
+      RestUtils.SubResource subResource, long blobSegmentNum) {
     this.prefix = prefix;
     this.clusterName = clusterName;
     this.pathAfterPrefixes = pathAfterPrefixes;
     this.operationOrBlobId = operationOrBlobId;
     this.subResource = subResource;
+    this.blobSegmentNum = blobSegmentNum;
```

Comment:
After reading the code in `GetBlobOptionsBuilder`,  I think we may change `blobSegmentNum` to `blobSegmentIndex`?  At first glance, I thought `blobSegmentNum` was number of blob segments.

Reply:
changing to `blobSegmentIdx`